tests/test_real_time_tools.cpp

```diff
@@ -270,4 +272,25 @@ TEST_F(TestRealTimeTools, test_spinner_loop_too_slow)
   double therotical_time_in_the_loop = period_sec * static_cast<double>(2 * nb_it - 1);
 
   ASSERT_NEAR(time_in_the_loop, therotical_time_in_the_loop, 0.1*period_sec);
-}
\ No newline at end of file
+}
+
+TEST_F(TestRealTimeTools, test_frequency_manager)
+{
+  double frequency = 200.0;
+  FrequencyManager freq_manager(frequency);
+  RealTimeCheck check(frequency,frequency*2);
+  bool success = true;
+  for(int i=0;i<400;i++)
+    {
+      check.tick();
+      success = freq_manager.wait();
+      ASSERT_EQ(success,true);
+    }
+  ASSERT_NEAR(frequency,check.get_average_frequency(),20);
+  for(int i=0;i<400;i++)
+    {
+      Timer::sleep_sec(0.006);
```

Comment:
Here this is supposed to fail because you wait longer than the time periode?
What happen then in the wait?
Do you wait at all?
What happen the next iteration?

Reply:
yes, it fails because it is impossible to both wait this long and enforce 200Hz. So in practice, the frequency manager never waits. And it returns false to inform that it fails to enforce 200Hz.